joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -37,44 +37,91 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
+    val name         = funcDecl.json(ParserKeys.Name).obj(ParserKeys.Name).str
+    val receiverInfo = getReceiverInfo(Try(funcDecl.json(ParserKeys.Recv)))
+    val (methodFullname, recSignStr) = receiverInfo match
+      case Some(_, typeFullName, evaluationStrategy, _) =>
+        val signatureStr = evaluationStrategy match
+          case EvaluationStrategies.BY_SHARING =>
+            s"*$typeFullName"
+          case EvaluationStrategies.BY_VALUE =>
+            typeFullName
+        (s"$typeFullName.$name", s"($signatureStr)")
+      case _ =>
+        (s"${fullyQualifiedPackage}.$name", "")
     // TODO: handle multiple return type or tuple (int, int)
     val genericTypeMethodMap = processTypeParams(funcDecl.json(ParserKeys.Type))
     val (returnTypeStr, methodReturn) =
       getReturnType(funcDecl.json(ParserKeys.Type), genericTypeMethodMap).headOption
         .getOrElse(("", methodReturnNode(funcDecl, Defines.voidTypeName)))
-    val templateParams = ""
-    val params         = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
+    val params = funcDecl.json(ParserKeys.Type)(ParserKeys.Params)(ParserKeys.List)
     val signature =
-      s"$fullname$templateParams(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
-    GoGlobal.recordFullNameToReturnType(fullname, returnTypeStr, Some(signature))
-    val methodNode_ = methodNode(funcDecl, name, funcDecl.code, fullname, Some(signature), filename)
+      s"$methodFullname$recSignStr(${parameterSignature(params, genericTypeMethodMap)})$returnTypeStr"
+    GoGlobal.recordFullNameToReturnType(methodFullname, returnTypeStr, Some(signature))
+    val methodNode_ = methodNode(funcDecl, name, funcDecl.code, methodFullname, Some(signature), filename)
     methodAstParentStack.push(methodNode_)
     scope.pushNewScope(methodNode_)
+    val receiverNode = astForReceiver(receiverInfo)
     val astForMethod =
       methodAst(
         methodNode_,
-        astForMethodParameter(params, genericTypeMethodMap),
+        receiverNode ++ astForMethodParameter(params, genericTypeMethodMap),
         astForMethodBody(funcDecl.json(ParserKeys.Body)),
         methodReturn
       )
     scope.popScope()
     methodAstParentStack.pop()
-//    if method is related to Struct then fill astParentFullName and astParentType
-    if (funcDecl.json.obj.contains(ParserKeys.Recv)) {
-      val receiverTypeDeclType =
-        parameterSignature(funcDecl.json(ParserKeys.Recv)(ParserKeys.List), Map.empty[String, List[String]])
-      methodNode_.astParentFullName = receiverTypeDeclType.replace("*", "")
-      methodNode_.astParentType = NodeTypes.TYPE_DECL
-      Ast.storeInDiffGraph(astForMethod, diffGraph)
-      Seq.empty
-    } else {
-      Seq(astForMethod)
-    }
+    receiverInfo match
+      case Some(_, typeFullName, _, _) =>
+        // if method is related to Struct then fill astParentFullName and astParentType
+        methodNode_.astParentFullName = typeFullName
```

Comment:
Could we try to change these properties via setter methods for consistency?
```scala
methodNode_.astParentFullName(typeFullName)
    .astParentType(NodeTypes.TYPE_DECL)
```

Reply:
done